### PR TITLE
✨ go.mod: Bump capi to main commit 4061a478b1610b308207da434e4eb4fc7af55de7n

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

* Bumps `sigs.k8s.io/cluster-api` to commit `0f47a19e038ee6b0d3b1e7675a62cdaf84face8c` to prepare for CAPI v1.7 release
* Removes `test/infrastructure/tmp-to-be-deleted` and use CAPI's package for inmemory instead

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: test/infrastructure/tmp-to-be-deleted/README.md
Comment (fabriziopandini) [inline test/infrastructure/tmp-to-be-deleted/README.md:1]: What about moving changes for dropping this folder in a separate commit (or in a follow up PR)

File: test/infrastructure/vcsim/controllers/controlplaneendpoint_controller.go [hunk 1]
```diff
@@ -23,15 +23,15 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/klog/v2"
+	inmemoryruntime "sigs.k8s.io/cluster-api/test/infrastructure/inmemory/pkg/runtime"
+	inmemoryserver "sigs.k8s.io/cluster-api/test/infrastructure/inmemory/pkg/server"
```
Comment (fabriziopandini) [inline test/infrastructure/vcsim/controllers/controlplaneendpoint_controller.go:27]: Should we move also those changes to another PR?

Comment (chrischdi) [reply test/infrastructure/vcsim/controllers/controlplaneendpoint_controller.go:27]: valid point 👍 🏃

**Review by fabriziopandini — COMMENTED** [review]
Thank you for driving this change!
/lgtm
/approve
